core/src/main/scala/com/salesforce/op/evaluators/OpForecastEvaluator.scala

```diff
@@ -62,40 +67,105 @@ private[op] class OpForecastEvaluator
   override def evaluateAll(data: Dataset[_]): ForecastMetrics = {
     val dataUse = makeDataToUse(data, getLabelCol)
 
-    val smape: Double = getSMAPE(dataUse, getLabelCol, getPredictionValueCol)
-    val metrics = ForecastMetrics(SMAPE = smape)
-
+    val metrics = getMetrics(dataUse, getLabelCol, getPredictionValueCol)
     log.info("Evaluated metrics: {}", metrics.toString)
     metrics
+
   }
 
-  protected def getSMAPE(data: Dataset[_], labelCol: String, predictionValueCol: String): Double = {
-    data.select(labelCol, predictionValueCol).rdd
-      .map(r => SMAPEValue(r.getAs[Double](0), r.getAs[Double](1)))
-      .reduce(_ + _).value
+
+  protected def getMetrics(data: Dataset[_], labelCol: String, predictionValueCol: String): ForecastMetrics = {
+    val res = data.select(labelCol, predictionValueCol).rdd
+      .map(r => MetricValue(r.getAs[Double](0), r.getAs[Double](1))).coalesce(1)
```

Comment:
why `coalesce(1)`? it would be very slow to execute.

Reply:
@tovbinm that is the whole point of my questions in the description. You have to run the fold from the first result to the last. Without the coalesce it seems to apply fold per partition (or at least this it my guess given the observed behavior and wrong result)